packages/fetchai/connections/p2p_noise/connection.py

```diff
@@ -0,0 +1,416 @@
+        raise e
+
+    return proc
+
+
+class Uri:
+    def __init__(
+        self,
+        uri: Optional[str] = None,
+        addr: Optional[str] = None,
+        port: Optional[int] = None,
+    ):
+        if uri is not None:
+            split = uri.split(":", 1)
+            self._addr = split[0]
+            self._port = split[1]
+        elif addr is not None and port is not None:
+            self._addr = addr
+            self._port = port
+        else:
+            raise ValueError("Either 'uri' or both 'addr' and 'port' must be set")
+
+    def __str__(self):
+        return "{}:{}".format(self._addr, self._port)
+
+    def __repr__(self):
+        return self.__str__()
+
+    @property
+    def addr(self) -> str:
+        return self._addr
+
+    @property
+    def port(self) -> int:
+        return self._port
+
+
+# TOFIX(LR) NOT thread safe
+class NoiseNode:
+    r"""Noise p2p node as a subprocess with named pipes interface
+    """
+
+    def __init__(
+        self,
+        key: PrivKey,
+        uri: Optional[Uri],
+        entry_peers: Sequence[Uri],
+        source: Union[Path, str],
+        clargs: Optional[List[str]] = [],
+    ):
+        """
+        Initialize a p2p noise node.
+
+        :param key: ec25519 curve private key.
+        :param uri: noise node ip address and port number in format ipaddress:port.
+        :param entry_peers: noise entry peers ip address and port numbers.
+        """
+
+        # node id in the p2p network
+        self.key = str(key)
+        self.pub = str(key.pub())
+
+        # node uri
+        self.uri = uri
+
+        # entry p
+        self.entry_peers = entry_peers
+
+        # node startup
+        self.source = source
+        self.clargs = clargs
+
+        # named pipes (fifos)
+        self.noise_to_aea_path = "/tmp/{}-noise_to_aea".format(self.pub[:5])
+        self.aea_to_noise_path = "/tmp/{}-aea_to_noise".format(self.pub[:5])
+        self._noise_to_aea = None
+        self._aea_to_noise = None
+        self._connection_attempts = 10
+
+        #
+        self._loop = None
+        self.proc = None
+
+    async def start(self) -> None:
+        if self._loop is None:
+            self._loop = asyncio.get_event_loop()
+
+        # get source deps
+        # TOFIX(LR) async version
+        # proc = await _async_golang_get_deps(self.source, loop=self._loop)
+        # await proc.wait()
+        proc = _golang_get_deps(self.source)
+        proc.wait()
+
+        # setup fifos
+        in_path = self.noise_to_aea_path
+        out_path = self.aea_to_noise_path
+        logger.debug("Creating pipes ({}, {})...".format(in_path, out_path))
+        if os.path.exists(in_path):
+            os.remove(in_path)
+        if os.path.exists(out_path):
+            os.remove(out_path)
+        os.mkfifo(in_path)
+        os.mkfifo(out_path)
+
+        # setup config
+        env = os.environ
+        env["ID"] = self.key + self.pub
```

Comment:
I am worried that these very generic Env variables might easily clash with existing ones. Can we have much more explicit ones, e.g. `AEA_P2P_...`